src/protocol.rs

```diff
@@ -481,7 +484,24 @@ mod tests {
             Err(panic) => {
                 if let Some(s) = panic.downcast_ref::<String>() {
                     println!("Panic occurred: {s}");
-                    assert_eq!(s, "Participant Config Size Error");
+                    assert_eq!(s, "Received a message with too few parties (Participant Config should have at least 2 parties)");
+                }
+            }
+        }
+
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let _p = ParticipantConfig::random_quorum(1, &mut rng);
+        });
+
+        match result {
+            Ok(_) => {
+                println!("No panic occurred.");
+            }
+            Err(panic) => {
+                if let Some(s) = panic.downcast_ref::<String>() {
+                    println!("Panic occurred: {s}");
+                    assert_eq!(s, "Received a message with too few parties (Participant Config should have at least 2 parties)");
```

Comment:
todo: this code is identical to L477 - 490 except for the number of parties. Can you make it a loop over those two values instead to reduce duplication?